src/commonMain/kotlin/fr/acinq/lightning/blockchain/electrum/ElectrumMiniWallet.kt

```diff
@@ -0,0 +1,162 @@
+package fr.acinq.lightning.blockchain.electrum
+
+import fr.acinq.bitcoin.*
+import fr.acinq.lightning.utils.Connection
+import fr.acinq.lightning.utils.lightningLogger
+import fr.acinq.lightning.utils.sat
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+import kotlin.native.concurrent.ThreadLocal
+
+data class WalletState(val addresses: Map<String, List<UnspentItem>>, val privateKeys: Map<String, PrivateKey>) {
```

Comment:
We can't really work with `UnspentItem` for dual funding, we need the whole previous transaction...
Can't we define the following class and add more requests to Electrum to get this information?

```kotlin
data class Utxo(val previousTx: Transaction, val outputIndex: Int, val height: Long) {
    val outPoint = OutPoint(previousTx, outputIndex.toLong())
    val amount = previousTx.txOut[outputIndex].amount
}
```

Reply:
Also, what does `height` mean? I'm not sure whether it's the number of confirmations or the block height at which this utxo was created, it's probably worth renaming to clarify.

Reply:
I did the bare minimum in 6b8209b. Notably, parent txs are not cleaned up.

Reply:
That's good enough, I don't think we care about cleaning up the parent txs.